aten/src/ATen/native/transformers/transformer.cpp

```diff
@@ -0,0 +1,136 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/NestedTensorImpl.h>
+
+#include <torch/library.h>
+
+#include <ATen/native/nested/NestedTensorTransformerFunctions.h>
+#include <ATen/native/transformers/library.h>
+#include <ATen/native/transformers/transformer.h>
+
+namespace at {
+
+namespace native {
+
+namespace {
+Tensor linear_for_ffn(
+    const Tensor& bias,
+    const Tensor& mat1,
+    const Tensor& mat2,
+    c10::optional<bool> use_gelu) {
+  if (mat1.is_nested()) {
+    return NestedTensor_times_Tensor_plus_Tensor_addmm(
+        bias, mat1, mat2.t(), 1, 1, use_gelu);
+  }
+
+  auto mat1_ = mat1.view({mat1.sizes()[0] * mat1.sizes()[1], mat1.sizes()[2]});
+  Tensor result;
+  if (use_gelu.has_value()) {
+    result = at::_addmm_activation(bias, mat1_, mat2.t(), 1, 1, *use_gelu);
+  } else {
+    result = at::addmm(bias, mat1_, mat2.t());
+  }
+  return result.view({mat1.sizes()[0], mat1.sizes()[1], -1});
+}
+} // namespace
+Tensor ffn_cpu(
+    const Tensor& input,
+    const Tensor& w1,
+    const Tensor& b1,
+    const Tensor& w2,
+    const Tensor& b2,
+    bool use_gelu,
+    bool add_norm) {
+  TORCH_CHECK(add_norm == false, "TODO add_norm to be supported in FFN");
```

Comment:
it's customary to ask people to report a bug if they want support hee hee ;)

Reply:
I guess this shouldn't actually get hit by the test for eligibility in the normal path